src/test/java/com/iexec/common/sdk/order/payload/RequestOrderTest.java

```diff
@@ -48,18 +63,47 @@ void shouldSerializeArgs() {
 
     @Test
     void shouldSerializeInputFiles() {
-        String expectedData = "{\"iexec_input_files\":[\"file1\",\"file2\"]}";
+        String expectedData = "{\"iexec_input_files\":[\"" + INPUT_FILE1 + "\",\"" + INPUT_FILE2 + "\"]}";
         DealParams params = DealParams.builder()
-                .iexecInputFiles(List.of("file1", "file2"))
+                .iexecInputFiles(List.of(INPUT_FILE1, INPUT_FILE2))
                 .build();
         Assertions.assertEquals(expectedData, RequestOrder.toStringParams(params));
     }
 
     @Test
     void shouldSerializeSecrets() {
-        String expectedData = "{\"iexec_secrets\":{\"0\":\"0\"}}";
+        Map<String, String> secrets = Map.of("0", "secretA", "2", "secretD");
+        String serializedSecrets = secrets.entrySet().stream()
+                .map(entry -> "\"" + entry.getKey() + "\":\"" + entry.getValue() + "\"")
+                .collect(Collectors.joining(","));
```

Comment:
I find this more readable:
```suggestion
        String serializedSecrets = "{\"0\":\"secretA\", \"2\":\"secretD\"}"
```

Reply:
Sadly, I don't have much of a choice here.
The keys are unordered in a Map.
Serialized data can thus be `{"0":"secretA","2":"secretD"}` or `{"2":"secretD","0":"secretA"}`
That's why I decided to perform my own serialization with a stream...

Reply:
I see. For that reason I generally prefer reading `expectedData` in a `JSONObject` and compare its fields.
No problem, we can keep it that way.

Reply:
What if you provide a `LinkedHashMap` instead? Its order should be predictable.
See https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html.

Reply:
Honestly, I don't really like to use `LinkedHashMap` for the tests when we use just `Map` everywhere.
I don't really like to use 2 json serialization implementation when we don't really care about the JSON strings.

I think what we really care is proving something like:
`stringToDealParams(toStringParams(dealParams)).equals(dealParams)`

`ChainDeal#toStringParams` is in fact `DealParams#toString`
`RequestOrder#toStringParams` is in fact `DealParams#parseString`

To provide the maximum value, I would prefer to do this kind of refactoring

Reply:
See 23794c6